draft-ietf-rtcweb-jsep.xml

```diff
@@ -1568,6 +1584,28 @@ Suggested:
           remote candidate, connectivity checks will be sent to the new
           candidate.</t>
         </section>
+        <section anchor="sec.onicecandidate" numbered="true" toc="default">
+          <name>onicecandidate Event</name>
+          <t>The onicecandidate event is dispatched to the application when the
+          ICE agent has discovered a new allowed local ICE candidate during ICE
+          gathering, as outlined in
+          <xref target="sec.ice-gather-overview" format="default"></xref> and 
+          subject to the restrictions discussed in 
+          <xref target="sec.ice-candidate-policy" format="default"></xref>.
+          The new candidate is supplied as an IceCandidate object 
+          (<xref target="sec.ice-candidate-format" format="default"/>) in 
+          the event, and all of its fields <bcp14>MUST</bcp14> be filled in. 
```

Comment:
What does "filled in" mean in this context. In a number of cases these will be "null". Do you just mean they won't be "undefined"?

Reply:
Well, when there is a new ICE candidate all fields will be non-null, which is what I am trying to say.

Reply:
I might be missing something, but as I read the text raddr/rport is null for host candidates, which should trigger onicecandidate, no?

Reply:
yes, but IceCandidate doesn't break those out into their own fields. IceCandidate (in JSEP) defines
- candidate (a la a=candidate)
- ufrag
- m-line index
- mid

all of which are filled in for all local candidates.

Reply:
(as outlined in https://tools.ietf.org/html/draft-ietf-rtcweb-jsep-24#section-3.5.2.1)

Reply:
Got it. I was thinking of rtcIceCandidate. Thankss for correcting me.